transport-native-epoll/src/main/c/netty_unix_util.c

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <string.h>
+#include "netty_unix_util.h"
+
+char* netty_unix_util_prepend(const char* prefix, const char* str) {
+    if (prefix == NULL) {
+        char* result = (char*) malloc(sizeof(char) * (strlen(str) + 1));
+        strcpy(result, str);
+        return result;
+    } else {
+        char* result = (char*) malloc(sizeof(char) * (strlen(str) + strlen(prefix) + 1));
```

Comment:
I'm not sure it is necessary. This method is called before the exception classes are loaded in `io_netty_channel_unix_Errors.c` so we can do `netty_unix_errors_throwOutOfMemoryError`. With the current code structure if we core dump in this condition that may be good enough.